components/hab/src/main.rs

```diff
@@ -409,6 +417,28 @@ fn sub_bldr_encrypt(ui: &mut UI, m: &ArgMatches) -> Result<()> {
     command::bldr::encrypt::start(ui, &url, &content, &default_cache_key_path(Some(&*FS_ROOT)))
 }
 
+fn sub_bldr_channel_create(ui: &mut UI, m: &ArgMatches) -> Result<()> {
+    let url = bldr_url_from_matches(m);
+    let origin = origin_param_or_env(&m)?;
+    let channel = m.value_of("CHANNEL").unwrap(); // Required via clap
+    let token = auth_token_param_or_env(&m)?;
+    command::bldr::channel::create::start(ui, &url, &token, origin, &channel)
+}
+
+fn sub_bldr_channel_destroy(ui: &mut UI, m: &ArgMatches) -> Result<()> {
+    let url = bldr_url_from_matches(m);
+    let origin = origin_param_or_env(&m)?;
+    let channel = m.value_of("CHANNEL").unwrap(); // Required via clap
+    let token = auth_token_param_or_env(&m)?;
+    command::bldr::channel::destroy::start(ui, &url, &token, origin, &channel)
+}
+
+fn sub_bldr_channel_list(ui: &mut UI, m: &ArgMatches) -> Result<()> {
+    let url = bldr_url_from_matches(m);
+    let origin = origin_param_or_env(&m)?;
```

Comment:
If you were to pass `&str` instead of `String` to those functions above as I mentioned, you could convert it to `&str` in these 3 functions.